presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/RemoteTable.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.plugin.jdbc;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Joiner;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class RemoteTable
+{
+    // catalog, schema and table names are reported by the remote database
+    private final Optional<String> catalogName;
+    private final Optional<String> schemaName;
+    private final String tableName;
+
+    @JsonCreator
+    public RemoteTable(
+            @JsonProperty Optional<String> catalogName,
+            @JsonProperty Optional<String> schemaName,
+            @JsonProperty String tableName)
```

Comment:
By convention, we use `@JsonProperty("xxxx")` (named params).
This may be redundant now since https://github.com/airlift/airbase/commit/d6de168d61adf041f8efb8ff29da63b0981c05c6 ...

However, then I am not sure whether `@JsonProperty` adds anything. Maybe it's totally redundant?

Reply:
BTW add JSON roundtrip test.

Reply:
`@JsonProperty` is not needed. Thanks for this idea. Good to know. Test was also nice.